repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/PowerFilter.java

```diff
@@ -350,72 +352,69 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         final long startTime = System.currentTimeMillis();
         long streamLimit = containerConfigurationService.getContentBodyReadLimit();
 
-        final MutableHttpServletRequest mutableHttpRequest = MutableHttpServletRequest.wrap((HttpServletRequest) request, streamLimit);
-        final MutableHttpServletResponse mutableHttpResponse = MutableHttpServletResponse.wrap(mutableHttpRequest, (HttpServletResponse) response);
+        final HttpServletRequestWrapper wrappedRequest = new HttpServletRequestWrapper((HttpServletRequest) request,
+                new BufferedServletInputStream(new LimitedReadInputStream(streamLimit, request.getInputStream())));
+        final HttpServletResponseWrapper wrappedResponse = new HttpServletResponseWrapper((HttpServletResponse) response,
+                ResponseMode.PASSTHROUGH,
```

Comment:
this seems wrong, don't we need to change the content related headers potentially in rms? I guess we could leave it like this and be sure to call the set methods in rms. Either way it's imperative that the response is uncommited before we hand off to rms.